runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/CommonCoderTest.java

```diff
@@ -278,41 +290,90 @@ private static Object convertValue(Object value, CommonCoder coderSpec, Coder co
       return WindowedValue.of(windowValue, timestamp, windows, paneInfo);
     } else if (s.equals(getUrn(StandardCoders.Enum.DOUBLE))) {
       return Double.parseDouble((String) value);
+    } else if (s.equals(getUrn(StandardCoders.Enum.ROW))) {
+      Schema schema;
+      try {
+        schema = SchemaTranslation.fromProto(SchemaApi.Schema.parseFrom(coderSpec.getPayload()));
+      } catch (InvalidProtocolBufferException e) {
+        throw new RuntimeException("Failed to parse schema payload for row coder", e);
+      }
+
+      return parseField(value, Schema.FieldType.row(schema));
     } else {
       throw new IllegalStateException("Unknown coder URN: " + coderSpec.getUrn());
     }
   }
 
+  private static Object parseField(Object value, Schema.FieldType fieldType) {
+    switch (fieldType.getTypeName()) {
+      case BYTE:
+        return ((Number) value).byteValue();
+      case INT16:
+        return ((Number) value).shortValue();
+      case INT32:
+        return ((Number) value).intValue();
+      case INT64:
+        return ((Number) value).longValue();
+      case FLOAT:
+        return Float.parseFloat((String) value);
+      case DOUBLE:
+        return Double.parseDouble((String) value);
```

Comment:
I tried to be consistent with the YAML representations used for other coders in `convertValues`. We use strings for `DoubleCoder` [there](https://github.com/apache/beam/blob/master/runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/CommonCoderTest.java#L280), presumably to avoid the possibility of running into precision errors? It looks like yaml does [support](https://yaml.org/type/float.html) floating point but explicitly states it doesn't specify a required accuracy for implementations. 

@lukecwik added the `parseDouble` line I linked in https://github.com/apache/beam/pull/8205, maybe he can clarify?

Reply:
YAML is JSON which supports doubles (really, as its only data type). @lukecwik I'm curious, but again this is not a blocker.

Reply:
Ok, I created [BEAM-8437](https://issues.apache.org/jira/browse/BEAM-8437) lets take this conversation over there.